### PR TITLE
[clang][Interp] Handle variadic functions

### PR DESCRIPTION
**Pull request by tbaederr**
Similarly to the code we already had for builtin functions, we need to check the call expression for the arguments passed.

### CODE REVIEW

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

**Comment by tbaederr** [comment]
Ping

File: clang/lib/AST/Interp/Interp.cpp [hunk 1]
```diff
@@ -121,18 +121,47 @@ static bool CheckGlobal(InterpState &S, CodePtr OpPC, const Pointer &Ptr) {
...
-  for (int32_t I = CE->getNumArgs() - 1; I >= 0; --I) {
-    const Expr *A = CE->getArg(I);
-    PrimType Ty = S.getContext().classify(A->getType()).value_or(PT_Ptr);
-    TYPE_SWITCH(Ty, S.Stk.discard<T>());
+  if (S.Current->Caller && CurFunc->isVariadic()) {
+    // CallExpr we're look for is at the return PC of the current function, i.e.
+    // in the caller.
+    // This code path should be executed very rarely.
+    const CallExpr *CE =
+        cast<CallExpr>(S.Current->Caller->getExpr(S.Current->getRetPC()));
```
Comment (AaronBallman) [inline clang/lib/AST/Interp/Interp.cpp:153]:
```suggestion
    const auto *CE =
        cast<CallExpr>(S.Current->Caller->getExpr(S.Current->getRetPC()));
```

**Review by AaronBallman — APPROVED** [review]
LGTM

**Pull request merged**